doc/ui-guidelines.md

```diff
@@ -0,0 +1,254 @@
+      :on-press-out        on-press-out}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
+
+```clojure
+(defn view
+  [{:keys [selected-tab]}]
+  (let [collectible-list (rf/sub [:wallet/all-collectibles])
+        on-collectible-press (rn/use-callback
+                              (fn [{:keys [id]}]
+                               (rf/dispatch [:wallet/get-collectible-details id])))]
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In certain instances, components utilized regular atoms; however, they should now be used with `rn/use-ref-atom`
+
+BEFORE:
+```clojure
+(defn view
+  []
+  (let [focused? (atom false)]
+    (fn []
+      (let [on-clear #(reset! status (if @focused? :active :default))
+            on-focus #(reset! focused? true)
+            on-blur  #(reset! focused? false)]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [focused? (rn/use-ref-atom false)
+        on-clear (rn/use-callback #(set-status (if @focused? :active :default)))
+        on-focus (rn/use-callback #(reset! focused? true))
+        on-blur  (rn/use-callback #(reset! focused? false))]))
+```
+
+## Effects
+
+LIFECYCLE:
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let [opacity (reanimated/use-shared-value 0)]
+    (rn/use-mount #(reanimated/animate opacity 1))
+    [rn/view
+     {:style (style/opacity opacity)}]))
+```
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let []
+    (rn/use-unmount #(rn/dispatch [:unmounted]))
+    [rn/view]))
+```
+
+> [!IMPORTANT]
+> Effects should NOT be utilized as a response to state changes for modifying logic. If you're unsure how to achieve this without using effects, please consult the team in the general chat. There may be instances where using effects is appropriate, so we can explore a solution together and enhance our guidelines.
+
+BAD:
+```clojure
+(defn f-zoom-button
+  [{:keys [selected? current-zoom]}]
+  (let [size (reanimated/use-shared-value (if selected? 37 25))]
+    (rn/use-effect #(reanimated/animate size (if selected? 37 25)) [current-zoom])
+    [rn/touchable-opacity
+     {:style (style/zoom-button-container size)}]))
+```
+
+BAD:
+
+```clojure
+(defn view
+  [collectible-list (rf/sub [:wallet/all-collectibles])]
+  (let []
+    (rn/use-effect #(rn/dispatch [:all-collectibles-changed]) [collectible-list])
+    [rn/view]))
+```
+
+Instead `:all-collectibles-changed` should be used in the handler which changes `collectible-list` state
+
```

Comment:
> Effects should NOT be utilized as a response to state changes for modifying logic

This is because Clojure data is immutable, right? So, custom hooks are not allowed as well unless written in JavaScript?

Reply:
custom hooks allow to separate the logic from the view though

Reply:
custom hooks allow to separate the logic from the view though